openapi/src/main/java/io/micronaut/openapi/visitor/AnnProcessorEnvironment.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.openapi.visitor;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import io.micronaut.context.ApplicationContextConfiguration;
+import io.micronaut.context.env.ActiveEnvironment;
+import io.micronaut.context.env.CachedEnvironment;
+import io.micronaut.context.env.DefaultEnvironment;
+import io.micronaut.context.env.Environment;
+import io.micronaut.context.env.PropertiesPropertySourceLoader;
+import io.micronaut.context.env.PropertySource;
+import io.micronaut.context.env.PropertySourceLoader;
+import io.micronaut.context.exceptions.ConfigurationException;
+import io.micronaut.core.io.ResourceLoader;
+import io.micronaut.core.io.file.DefaultFileSystemResourceLoader;
+import io.micronaut.core.order.OrderUtil;
+import io.micronaut.core.util.StringUtils;
+import io.micronaut.inject.visitor.VisitorContext;
+
+import static io.micronaut.openapi.visitor.OpenApiApplicationVisitor.MICRONAUT_CONFIG_FILE_LOCATIONS;
+
+/**
+ * Specific environment for annotation processing level. Solve problem with access to resources
+ * from project classpath.
+ *
+ * @since 4.5.0
+ */
+public class AnnProcessorEnvironment extends DefaultEnvironment {
+
+    private final List<String> annotationProcessingConfigLocations;
+    private final String projectResourcesPath;
+    private final String projectDir;
+
+    /**
+     * Construct a new environment for the given configuration.
+     *
+     * @param configuration The configuration
+     * @param context visitor context
+     */
+    public AnnProcessorEnvironment(ApplicationContextConfiguration configuration, VisitorContext context) {
+        super(configuration);
+
+        Path projectPath = context.getProjectDir().orElse(Paths.get(System.getProperty("user.dir")));
+        projectDir = projectPath.toString().replaceAll("\\\\", "/");
```

Comment:
this seems like a hack, why can't you just use `path.toUri()`?

Reply:
fixed

Reply:
@graemerocher Don't agree with you. This a stnadrad fix for windows. Why didn't I use `toUri` ? Let's see to example. My directory is `D:\Work\my-project` in windows and `/home/my/work/my-project`. I need to set this path as `file:D:\Work\my-project` for wind and `file:/home/my/work/my-project`.

If I use `path.toUri().toString()` than I have these paths: `file:///D:/Work/my-project/` for Windows and  `file:///D:/Work/my-project/` and `file:///home/my/work/my-project/`.

The problem is that in order to correctly display the correct path, I need to trim file and 3 slashes for Windows, and file and 2 slashes for Linux. In general, there are more problems than benefits from such a solution. My solution is simple and clear